Firebase/Core/Public/FIRConfiguration.h

```diff
@@ -47,21 +28,12 @@ NS_ASSUME_NONNULL_BEGIN
 NS_SWIFT_NAME(FirebaseConfiguration)
 @interface FIRConfiguration : NSObject
 
-#if defined(__IPHONE_10_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_10_0
```

Comment:
Class properties aren't an Xcode thing, they're an iOS thing. This effectively makes our minimum required iOS 10, which doesn't seem right.

Reply:
I was under the impression they were an Objective-C feature that was integrated into Xcode 8 -
 https://useyourloaf.com/blog/objective-c-class-properties/

Reply:
"Note that since this is a feature of the LLVM compiler in Xcode 8 it will work with deployment targets earlier than iOS 10"

Reply:
Interesting! I had read about this at the time and didn't realize that it was actually a compiler mechanism.